src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -197,25 +197,11 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                         {
                             _arguments.Append('"');
                             // need to escape all trailing backslashes so the native command receives it correctly
-                            if (arg.EndsWith('\\'))
-                            {
-                                // find index of last character that isn't a backslash
-                                var chars = arg.ToCharArray();
-                                int index = 0;
-                                for (int i = chars.Length-1; i > 0; i--)
-                                {
-                                    if (chars[i] != '\\')
-                                    {
-                                        index = i;
-                                        break;
-                                    }
-                                }
-                                _arguments.Append(arg.Substring(0, index));
-                                _arguments.Append(arg.Substring(index).Replace("\\","\\\\"));
-                            }
-                            else
+                            // according to http://www.daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULESDOC
+                            _arguments.Append(arg);
+                            for (int i = arg.Length-1; i >= 0 && arg[i] == '\\'; i--)
                             {
-                                _arguments.Append(arg);
+                                _arguments.Append('\\');
```

Comment:
Shouldn't it be "\\\\"?

Reply:
No, unless you want to triplicate trailing `\`. `arg` already have one copy of `\`. This just adding second copy on top of it.

Reply:
Thanks for clarify!